python/whylogs/api/writer/local.py

```diff
@@ -15,16 +20,26 @@ def __init__(self, base_dir: Optional[str] = None, base_name: Optional[str] = No
         self._base_dir = base_dir
         self._base_name = base_name
 
-    def write(self, profile: DatasetProfileView, dest: Optional[str] = None) -> None:
-        if dest is None:
+    def write(
+        self,
+        file: Optional[Writable] = None,
+        profile: Optional[DatasetProfileView] = None,
+        dest: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        if profile:
+            logger.warning("`profile will be deprecated, you should use `file` instead")
+            file = profile
+        if isinstance(file, HTMLReport) and dest is None:
```

Comment:
`Writable`-specific default destination logic should live in the `Writable`

Reply:
good point. not sure it will work for DatasetProfileView though, since each writer have a minor specification to where they will store the profile. i.e.: s3 treats the paths with `prefix` and `object_name`, while local works with `base_dir` and `file_name`. but i agree it should be standard

Reply:
yea, so it's a bit harder than i thought, specially with how mlflow writer's built at the moment. in order to write the artifact, we use a helper `_get_temp_directory` method, where we have to know beforehand the "proper" path to write to a `tempfile.NamedTemporaryFile`. it is doable for most cases, but the thing is that specially for mlflow, `file.write()` expects a valid path to be there and ready.

Reply:
happy to read your thoughts on this @richard-rogers

Reply:
`Writable` should have a method to return the path to the data the `Writer` should send to it's destination. Any default logic could be handled there.